evm-contracts/src/v0.6/dev/HistoricDeviationValidator.sol

```diff
@@ -0,0 +1,117 @@
+pragma solidity 0.6.6;
+
+import './AnswerValidatorInterface.sol';
+import '../Flags.sol';
+import '../Owned.sol';
+import '../vendor/SafeMath.sol';
+
+contract HistoricDeviationValidator is Owned, AnswerValidatorInterface {
+  using SafeMath for uint256;
+
+  uint32 constant public THRESHOLD_MULTIPLIER = 100000;
+
+  uint32 public flaggingThreshold;
+  Flags public flags;
+
+  event FlaggingThresholdUpdated(
+    uint24 indexed previous,
+    uint24 indexed current
+  );
+  event FlagsAddressUpdated(
+    address indexed previous,
+    address indexed current
+  );
+
+  constructor(
+    address newFlags,
+    uint24 newFT
+  )
+    public
+  {
+    setFlagsAddress(newFlags);
+    setFlaggingThreshold(newFT);
+  }
+
+  function validate(
+    uint256 previousRoundId,
+    int256 previous,
+    uint256 currentRoundId,
+    int256 current
+  )
+    external
+    override
+    returns (bool)
+  {
+    if (previous == 0) return true;
+
+    if (percentDiff(previous, current) > flaggingThreshold) {
+      flags.raiseFlags(arrayifyMsgSender());
```

Comment:
I am not particularly fond of the `arrayifyMsgSender` function name. As it's private and only used in this one place, why do not we just build an array inline?

```suggestion
      flags.raiseFlags([address(msg.sender)]);
```

Reply:
```
error HistoricDeviationValidator.sol:48:24: TypeError: Invalid type for argument in function call. Invalid implicit conversion from address[1] memory to address[] memory requested.
      flags.raiseFlags([address(msg.sender)]);
                       ^-------------------^
```

The two lines in that method were the only way I knew how to do it. Given it's clunkiness and it's just a type conversion, I pulled it out to make logic flow more clear. Can put it back if you feel strongly, or I'm open other function names if that's your biggest issue.

Reply:
Aha, Solidity dynamic vs. fixed arrays. Sorry, missed that. In that case, I do not have any better suggestions here.